### PR TITLE
Feat/osirix

### PR DESCRIPTION
**Pull request by OzgeYurtsever**

### CODE REVIEW

File: plugins/Other.js [hunk 1]
```diff
@@ -739,6 +740,92 @@ async function other(fastify) {
...
+                            })
+                        );
+                      });
+                      Promise.all(aimsavePromises)
+                        .then(() => {
+                          try {
+                            const uploadMsg = 'Upload Failed as ';
+                            let errMessage = 'none of the files were uploaded successfully';
+                            if (result.errors.length === promiseArr.length) {
+                              reject(new InternalError(uploadMsg, { message: errMessage }));
```
Comment (emelalkim) [inline plugins/Other.js:788]: it should be new Error(message)

File: plugins/Other.js [hunk 2]
```diff
@@ -739,6 +740,92 @@ async function other(fastify) {
...
+                            } else {
+                              // eslint-disable-next-line no-lonely-if
+                              if (nonSupported.length) {
+                                errMessage = `Not supported shapes in: ${nonSupported.join(', ')}`;
+                                const error = new InternalError('Upload completed with errors', {
+                                  message: errMessage,
+                                });
```
Comment (emelalkim) [inline plugins/Other.js:795]: new Error(message)

File: plugins/Other.js [hunk 3]
```diff
@@ -739,6 +740,92 @@ async function other(fastify) {
...
+                                if (result.errors.length > 0) {
+                                  const combinedErrors = result.errors.push(error);
+                                  fastify.log.info(`Saving successful`);
+                                  resolve({ success: true, errors: [combinedErrors] });
```
Comment (emelalkim) [inline plugins/Other.js:799]: either result.errors.concat(error)
or
[...result.errors, error]

File: plugins/Other.js [hunk 4]
```diff
@@ -739,6 +740,92 @@ async function other(fastify) {
...
+                                } else {
+                                  fastify.log.info(`Saving successful`);
+                                  resolve({ success: true, errors: [error] });
+                                }
+                              } else {
+                                fastify.log.info(`Saving successful`);
+                                resolve({ success: true, errors: [result.errors] });
```
Comment (emelalkim) [inline plugins/Other.js:806]: result.errors not []

File: plugins/Other.js [hunk 5]
```diff
@@ -804,6 +891,234 @@ async function other(fastify) {
       })
   );
 
+  fastify.decorate('createAimJsons', filteredOsirixArr => {
+    try {
+      const aimJsons = [];
+      filteredOsirixArr.forEach(el => {
+        const aim = new Aim(el.seedData, fastify.enumAimType.imageAnnotation);
```
Comment (emelalkim) [inline plugins/Other.js:898]: enum should be used from aimapi

File: plugins/Other.js [hunk 6]
```diff
@@ -804,6 +891,234 @@ async function other(fastify) {
...
+        LengthCm,
+        AreaCm2,
+      },
+      shapeIndex: IndexInImage,
+      type: Type,
+    };
+  });
+
+  fastify.decorate('createAimMarkups', (aim, markupsToSave) => {
+    // eslint-disable-next-line no-unused-vars
```
Comment (emelalkim) [inline plugins/Other.js:937]: seems redundant

File: plugins/Other.js [hunk 7]
```diff
@@ -804,6 +891,234 @@ async function other(fastify) {
...
+
+  // eslint-disable-next-line consistent-return
+  fastify.decorate('getTemplateAnswers', (metadata, annotationName, tempModality) => {
+    if (metadata.series) {
+      const { number, description, instanceNumber } = metadata.series;
+      const seriesModality = metadata.series.modality;
+      const comment = {
+        value: `${seriesModality} / ${description} / ${instanceNumber} / ${number}`,
+      };
+      // TODO: handle modality and suv hu
```
Comment (emelalkim) [inline plugins/Other.js:1085]: remove todo

File: plugins/Other.js [hunk 8]
```diff
@@ -804,6 +891,234 @@ async function other(fastify) {
...
+        },
+      ];
+      return { comment, modality, name, typeCode };
+    }
+  });
+
+  fastify.decorate('filterOsirixAnnotations', imagesArr => {
+    const filteredOsirix = {};
+    const nonSupported = [];
+    const supportedTypes = [19, 5, 14, 15, 10, 9, 28, 20, 11, 6];
```
Comment (emelalkim) [inline plugins/Other.js:1102]: write comment saying if new supported objects are added createAimMarkups should be updated

File: plugins/Other.js [hunk 9]
```diff
@@ -804,6 +891,234 @@ async function other(fastify) {
...
+        points,
+        LengthCm,
+        AreaCm2,
+      },
+      shapeIndex: IndexInImage,
+      type: Type,
+    };
+  });
+
+  fastify.decorate('createAimMarkups', (aim, markupsToSave) => {
```
Comment (emelalkim) [inline plugins/Other.js:936]: write comment saying if new supported objects are added filterOsirixAnnotations should be updated

File: plugins/Other.js [hunk 10]
```diff
@@ -804,6 +891,234 @@ async function other(fastify) {
...
+    // eslint-disable-next-line no-unused-vars
+
+    markupsToSave.forEach(value => {
+      const { type, markup, shapeIndex, imageReferenceUid } = value;
+      // eslint-disable-next-line default-case
+      // eslint-disable-next-line default-case
+      switch (type) {
```
Comment (emelalkim) [inline plugins/Other.js:943]: shapes ={19: point, 5:line, 14:line,15: polygon...}
supportedtypes array include => lookup
here shapes(type) 
case 'point':...

File: plugins/Other.js [hunk 11]
```diff
@@ -804,6 +891,234 @@ async function other(fastify) {
...
+          }
+        } else {
+          nonSupported.push(item.Name);
+        }
+        return all;
+      }, filteredOsirix);
+    });
+    return { filteredOsirix, nonSupported };
+  });
+
```
Comment (emelalkim) [inline plugins/Other.js:1121]: add sanity test

File: plugins/Other.js (hunk 1 above)
Comment (OzgeYurtsever) [reply plugins/Other.js:788]: Done!

File: plugins/Other.js (hunk 2 above)
Comment (OzgeYurtsever) [reply plugins/Other.js:795]: Done!

File: plugins/Other.js (hunk 3 above)
Comment (OzgeYurtsever) [reply plugins/Other.js:799]: Done!

File: plugins/Other.js (hunk 4 above)
Comment (OzgeYurtsever) [reply plugins/Other.js:806]: Done!

File: plugins/Other.js (hunk 5 above)
Comment (OzgeYurtsever) [reply plugins/Other.js:898]: Filed improvement ticket EPAD-3170.

File: plugins/Other.js (hunk 6 above)
Comment (OzgeYurtsever) [reply plugins/Other.js:937]: Done!

File: plugins/Other.js (hunk 7 above)
Comment (OzgeYurtsever) [reply plugins/Other.js:1085]: Done!

File: plugins/Other.js (hunk 8 above)
Comment (OzgeYurtsever) [reply plugins/Other.js:1102]: Done!

File: plugins/Other.js (hunk 9 above)
Comment (OzgeYurtsever) [reply plugins/Other.js:936]: Done!

File: plugins/Other.js (hunk 10 above)
Comment (OzgeYurtsever) [reply plugins/Other.js:943]: Filed ticket  EPAD-3171 for version 4.2

File: plugins/Other.js (hunk 11 above)
Comment (OzgeYurtsever) [reply plugins/Other.js:1121]: Filed ticket EPAD-3172 for version 4.1

**Pull request merged**